src/components/Modal.jsx

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable */
+
+import { useState, useRef } from 'react';
+import { useOnClickOutside } from '../useOnClickOutside';
+
+const Modal = () => {
+  const [openModal, setOpenModal] = useState(false);
+  const ref = useRef();
+
+  useOnClickOutside(ref, openModal, () => setOpenModal(false));
+  return (
+    <div className="modal">
+      <button onClick={() => setOpenModal(true)}>Modal</button>
+      {openModal && (
+        <div ref={ref} className="modalContent">
+          <span onClick={() => setOpenModal(false)}>X</span>
+          <div>Modal content here</div>
+        </div>
+      )}
+    </div>
+  );
```

Comment:
- [ ] It's great you implemented the modal component, and used a custom hook but kindly move the state `openModal` up to the parent component `TodoLoigc` so that other child components such as the `TodoItem` would be able to access